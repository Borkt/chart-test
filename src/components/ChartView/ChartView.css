.ChartView {
  align-items: center;
  flex: 3;
  color: #09d3ac;
  display: flex;
  flex-direction: column;
  font-size: calc(10px + 2vmin);
  height: 40vmin;
  justify-content: center;
  min-height: 100vh;
  text-align: center;
}
