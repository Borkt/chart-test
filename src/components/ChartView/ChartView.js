import React, { useState, useEffect } from 'react';
import PropTypes from 'prop-types';
import { TimeSeries } from "pondjs";
import {
  ChartContainer,
  ChartRow,
  Charts,
  Legend,
  LineChart,
  styler,
  YAxis,
} from "react-timeseries-charts";

import './ChartView.css';

export const ChartView = ({
  activeCampaignFilters,
  activeDatasourceFilters,
  data,
}) => {
  const [chartTitle, setChartTitle] = useState([]);

  useEffect(() => {
    const generateTitle = () => {
      let campaignTitle = " and Metrics: ";
      let datasourceTitle = "Datasource: "

      if (activeCampaignFilters.length > 0) {
        campaignTitle += activeCampaignFilters.map(f => f.value).join(", ");
      } else {
        campaignTitle += "All Campaigns";
      }

      if (activeDatasourceFilters.length > 0) {
        datasourceTitle += activeDatasourceFilters.map(f => f.value).join(", ") + " ";
      } else {
        datasourceTitle += "All ";
      }

      // return datasourceTitle + " / " + campaignTitle;
      setChartTitle(datasourceTitle + campaignTitle);
    }

    generateTitle();
  }, [activeCampaignFilters, activeDatasourceFilters]);


  if (data.length === 0) {
    return (
      <div className="ChartView">
        Data Loading or Unavailable...
      </div>
    );
  }

  const series = new TimeSeries({
    name: "Values",
    columns: ["time", "clicks", "impressions"],
    points: data,
  });

  const chartStyler = styler([
    {key: "clicks", color: "#2ca02c", width: 1},
    {key: "impressions", color: "#9467bd", width: 1}
  ]);


  const legendStyler = styler([
    {key: "clicks", color: "#2ca02c", width: 3 },
    {key: "impressions", color: "#9467bd", width: 3},
  ]);

  return (
    <div className="ChartView">
      <ChartContainer
        timeRange={series.range()}
        title={chartTitle}
        titleStyle={{ fill: "#555", fontWeight: 500, fontSize: "24px" }}
        width={900}
      >
        <ChartRow height="400">
            <YAxis
              id="axis1"
              label="Clicks"
              max={series.max("clicks")}
              min={0}
              type="linear"
              width="100"
            />
            <Charts>
                <LineChart
                  axis="axis1"
                  columns={["clicks"]}
                  series={series}
                  style={chartStyler}
                />
                <LineChart
                  axis="axis2"
                  columns={["impressions"]}
                  series={series}
                  style={chartStyler}
                />
            </Charts>
            <YAxis
              id="axis2"
              label="Impressions"
              max={series.max("impressions")}
              min={0}
              type="linear"
              width="100"
            />
        </ChartRow>
      </ChartContainer>

      <Legend
        type="swatch"
        align="right"
        style={legendStyler}
        categories={[
            { key: "clicks", label: "Clicks" },
            { key: "impressions", label: "Impressions" }
        ]}
      />
    </div>
  );
}

ChartView.propTypes = {
  activeCampaignFilters: PropTypes.array.isRequired,
  activeDatasourceFilters: PropTypes.array.isRequired,
  data: PropTypes.array.isRequired,
};
