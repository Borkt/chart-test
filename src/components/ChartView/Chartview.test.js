import React from 'react';
import { cleanup, render } from '@testing-library/react';
import { ChartView } from './ChartView';

// import { mockData } from '../../mock-data/mockData';

const defaultInitialProps =  {
  activeFilterOptions: {
    activeCampaignFilters: [],
    activeDatasourceFilters: [],
  },
  data: [],
};

// const propsWithMockedData = {
//   activeFilterOptions: mockData.activeFilterOptions,
//   data: mockData.data,
// };

afterEach(cleanup);

// Internal Chart functionality is not tested because that is
// handled internally by the 'react-timeseries-charts' package

describe('ChartView', () => {
  it('matches previous Snapshot', () => {
    const { asFragment } = render(<ChartView { ...defaultInitialProps } />);
    expect(asFragment()).toMatchSnapshot();
  });

  it('renders Chart initial text', () => {
    const { getByText } = render(<ChartView { ...defaultInitialProps } />);
    expect(getByText('Data Loading or Unavailable...')).toBeInTheDocument();
  });

  // it('renders Chart onload text', async () => {
  //   await act(async () => {
  //     const { getByText } = render(<ChartView { ...propsWithMockedData } />);
  //     expect(getByText('Datasource: All and Metrics: All Campaigns')).toBeInTheDocument();
  //   })
  // });
});
