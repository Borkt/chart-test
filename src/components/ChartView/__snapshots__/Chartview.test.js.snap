// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ChartView matches previous Snapshot 1`] = `
<DocumentFragment>
  <div
    class="ChartView"
  >
    Data Loading or Unavailable...
  </div>
</DocumentFragment>
`;
